hibike/hibike_message.py

```diff
@@ -112,31 +131,195 @@ def checksum(data):
 def send(serial_conn, message):
   m_buff = message.toByte()
   chk = checksum(m_buff)
-  m_buff.append(chr(chk))
+  m_buff.append(chk)
   encoded = cobs_encode(m_buff)
   out_buf = bytearray([0x00, len(encoded)]) + encoded
   serial_conn.write(out_buf)
 
 
-def make_sub_request(delay):
-  """ Makes and returns SubscriptionRequest message."""
-  temp_payload = struct.pack('<H', delay)
+
+def make_ping():
+  """ Makes and returns Ping message."""
+  payload = bytearray()
+  message = HibikeMessage(messageTypes["Ping"], payload)
+  return message
+
+def make_subscription_request(device_id, params, delay):
+  """ Makes and returns SubscriptionRequest message.
+
+      looks up config data about the specified 
+      device_id to properly construct the message.
+      
+      device_id - a device type id (not uid).
+      params    - an iterable of param names
+      delay     - the delay in milliseconds
+      struct.pack('%sf' % len(floatlist), *floatlist)
+  """
+  paramNums = [paramMap[device_id][name][0] for name in params]
```

Comment:
resolved, all bitmask operations have been abstracted into encode_params and decode_params